src/main/java/com/jagrosh/jmusicbot/audio/AudioHandler.java

```diff
@@ -59,12 +58,20 @@ public class AudioHandler extends AudioEventAdapter implements AudioSendHandler
     private final long guildId;
     
     private AudioFrame lastFrame;
+    private AbstractQueue<QueuedTrack> queue;
 
     protected AudioHandler(PlayerManager manager, Guild guild, AudioPlayer player)
     {
         this.manager = manager;
         this.audioPlayer = player;
         this.guildId = guild.getIdLong();
+
+        this.setQueueType(manager.getBot().getSettingsManager().getSettings(guildId).getQueueType());
+    }
+
+    public void setQueueType(QueueType type)
+    {
+        queue = (AbstractQueue<QueuedTrack>) type.createInstance(queue);
```

Comment:
@MichailiK I currently cannot think of any way to get rid of this cast

Reply:
Hm, I believe that we will need to create our own functional interface. We can make a `QueueSupplier` in the utils package:
```java
@FunctionalInterface
public interface QueueSupplier
{
    <T extends Queueable> AbstractQueue<T> apply(AbstractQueue<T> previous);
}
```

Then, we can go in our queuetype and replace all the `Function<...>`s with `QueueSupplier`. Then you can finally remove this cast.

Reply:
That seems to have done the trick, thanks ^^